hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/handle/validation/ExpiryValidatorImpl.java

```diff
@@ -100,7 +120,9 @@ public ExpiryMeta resolveCreationAttempt(
     @NonNull
     @Override
     public ExpiryMeta resolveUpdateAttempt(
-            @NonNull final ExpiryMeta currentMeta, @NonNull final ExpiryMeta updateMeta) {
+            @NonNull final ExpiryMeta currentMeta,
+            @NonNull final ExpiryMeta updateMeta,
+            final boolean isForTokenUpdate) {
```

Comment:
The other method of differentiating with the functionality enum seems easier to read and reason about, as opposed to the boolean added here. Any reason not to use functionality instead?

Reply:
We really only care whether the functionality enum is TOKEN_UPDATE or anything else. There are places in the smart contract code where it is known that some kind of smart contract functionality is in use but we don't specifically know which kind. In that case, it is clearer to simply pass 'false' (for not a TOKEN_UPDATE) versus either modifying a bunch of code to pass the exact smart contract functionality even though it doesn't ultimately matter, or passing a smart contract enum which is technically incorrect.